From d97c675b34ceb1077d81a1a0ee90e9c201489989 Mon Sep 17 00:00:00 2001
From: Jonathan Nieder <jrnieder@uchicago.edu>
Date: Wed, 28 May 2008 14:53:38 -0500
Subject: [PATCH] libgpm: do not log debug messages

This change puts src/lib/report-lib.c more in line
with src/report.c by suppressing debug messages by
default (although here, the only way to change that
default is to change the code and recompile :).

This is particularly important to users of logcheck,
which for some reason considers any syslog message
containing the text "debug" to represent a
misconfigured system and a security risk.

diff --git a/src/lib/report-lib.c b/src/lib/report-lib.c
index 812fdb8..c0ae086 100644
--- a/src/lib/report-lib.c
+++ b/src/lib/report-lib.c
@@ -29,6 +29,9 @@ void gpm_report(int line, char *file, int stat, char *text, ... )
    char *string = NULL;
    int log_level;
    va_list ap;
+
+   if (stat == GPM_STAT_DEBUG) return;
+
    va_start(ap,text);
 
    switch(stat) {
